Layout Instability API updates for Chrome 77 (#1171)
GoogleChrome/web.dev: src/site/content/en/blog/layout-instability-api/index.md

  }
})

observer.observe({type: 'layout-shift', buffered: true});
```

> <div class="w-aside w-aside--note">

Reviewer: ```suggestion
{% Aside %}
```

> **Note:** The

Reviewer: ```suggestion
The
```

Author: Per our chat discussion, I believe we decided to leave this in.

[`buffered`](https://w3c.github.io/performance-timeline/#dom-performanceobserverinit-buffered)
flag in the above example (supported in Chrome 77+) gives you access to entries
that may have occurred prior to creating the `PerformanceObserver`.

> </div>

Reviewer: ```suggestion
{% endAside %}
```

> <div class="w-aside w-aside--warning">

Reviewer: ```suggestion
{% Aside 'caution' %}
```

> **Beware:** The `entryType` value for this API has changed a few times during

Reviewer: ```suggestion
The `entryType` value for this API has changed a few times during
```

> the experimentation period. In Chrome 76 it was `layoutShift` and in Chrome

Reviewer: ```suggestion
the experimentation period. In Chrome 76 it was `layoutShift`, and in Chrome
```

74-75 it was `layoutJank`. Developers implementing the stable API should only
need to observe the current `layout-shift` value (as shown in the example
above), but developers who are part of the origin trial may need to observe
multiple entry types to cover their full user base. See [this
demo](https://output.jsbin.com/zajamil/quiet) for an example of code that
works in Chrome 74+.

> </div>

Reviewer: ```suggestion
{% endAside %}
```

  }
})

observer.observe({type: 'layout-shift', buffered: true});
```

<div class="w-aside w-aside--note">

**Note:** The
[`buffered`](https://w3c.github.io/performance-timeline/#dom-performanceobserverinit-buffered)
flag in the above example (supported in Chrome 77+) gives you access to entries

> that may have occurred prior to creating the performance observer.

Reviewer 2: ```suggestion
that may have occurred prior to creating the PerformanceObserver.
```

</div>

<div class="w-aside w-aside--warning">

**Beware:** The `entryType` value for this API has changed a few times during
the experimentation period. In Chrome 76 it was `layoutShift` and in Chrome
74-75 it was `layoutJank`.

Developers implementing the stable API should only need to observe the current
`layout-shift` value (as shown in the example above), but developers who are
part of the origin trial may need to observe multiple entry types to cover
their full user base).

```js
if (PerformanceObserver.supportedEntryTypes &&
    PerformanceObserver.supportedEntryTypes.includes('layout-shift')) {
  observer.observe({type: 'layout-shift', buffered: true});
} else {

>   observer.observe({entryTypes: ['layoutShift', 'layoutJank']});

Reviewer 3: Should we include the use of getEntriesByType for the version that doesn't have the buffered flag?

Author: Hmmm, that would make the example quite a bit longer (since we'd have to include the original callback). What about just linking to your example (assuming that's not going to be deleted)?

Reviewer 3: That sounds fine.
